### ISSUE TITLE
Catmandu::Importer::CSV should handle header-less data

### ISSUE THREAD

**Issue by nichtich** [LibreCat/Catmandu#144]
echo test | catmandu convert CSV --header 0
    Oops! EHR - getline_hr () called before column_names () at /home/voj/perl5/perlbrew/perls/perl-5.18.0/lib/site_perl/5.18.0/Catmandu/Importer/CSV.pm line 54.

Expected output similar to:

    {"A":"test"}

with an automatically generated column names such as in Excel.

**Comment by jorol** [comment]
What would be the best strategy to determine the number of columns if `header` or `fields` are not defined? I've tried <https://github.com/jorol/Catmandu/blob/gh%23144/lib/Catmandu/Importer/CSV.pm#L45> to get it from the first CSV line and to "rewind" the filehandle. This works for files, but not with piped arguments like `echo test | catmandu convert CSV --header 0`. An alternative would be to determine it within the `generator` sub for each row and construct the hash manually:

```perl
...
my $line = $self->csv->getline($fh) // return;
my %hash;
@hash{ 'A' .. _int2col(scalar @{$line} - 1) } = @{$line};
...
```

Any recommendations?

**Comment by nics** [comment]
Hi Johann,

i’m almost finished with a collect_fields option for these tabular exporters that collects all keys first. Downside of this approach is that all data is buffered in memory when collect_fields is true. We can't rely on all io handles to be rewindable. And if they are, they might return different results when reread.

**Comment by nics** [comment]
we don't need to know the number of columns when converting to a hash. The exporter has that problem, but not the importer.

**Comment by nics** [comment]
implemented in 821ef0f867702255aea002ccd9ef9c766f125b10. The generated field names are the column indexes (0,1.2,...). We can add the excel convention (A,B,C) later as an option.
